### DISCUSSION TITLE
List of last items used

### DISCUSSION THREAD

**Discussion by ThomasAusNRW** [logisim-evolution/logisim-evolution#1694] [Ideas]
It would be nice if you can implement a list of the last used items. My idea is to use a context-menu by clicking the right mouse-button on an empty place. The list should contain a certain number of different items, for example  the last eight ones.

**Comment by maehne** [comment]
Thanks for the proposal. It could be useful. I think it would make sense to make the length of the list configurable. One might also think about whether adding the possibility to mark components from different libraries as favourites would be more useful. IMHO, the natural place for such lists (most recently used, favourites) would be in the library pane itself. However, I can see the point to make it accessible via a context menu to speed up access to them. It would be nice if more users would express their opinion about such functionality. The new feature(s) need also to be refined to describe more precisely the purpose and how it should behave / be used. PRs in that direction are welcome.

**Reply by juli-99** [reply]
> One might also think about whether adding the possibility to mark components from different libraries as favourites would be more useful.

Doesn't the toolbar already provide this functionality?
It's in build configuration lacks a lot of features. (`Project >> Options... >> Toolbar`)
Because of this, it can only be configured effectively with the help of editing the `.circ`/`template` file.

Instead of adding another feature, I'm in favor of improving the existing ones.

**Reply by juli-99** [reply]
Adding placeholders that represent the last n used components to the toolbar might be a good solution.
